src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,204 @@
+const FRAMERATE = 60;
+const INTERVAL_TIME = 1000 / FRAMERATE;
+// if is true console.log are enabled to sniff animation'state
+const debugAnimation = false;
+
+// player statut
+const PLAYER_STATE = {
+    // player is stopped
+    STOP: 0,
+    // player plays animation
+    PLAY: 1,
+    // player is at the end of an animation
+    END: 2,
+    // player is paused
+    PAUSE: 3,
+};
+
+const debugMsg =
+    [
+        'Stop',
+        'Play',
+        'End',
+        'Pause',
+    ];
+
+// if debugAnimation is true console.log are enabled to sniff animation'state
+let _DEBUG = null;
+
+if (debugAnimation) {
+    _DEBUG = function DEBUG(message, animation) {
+        if (animation) {
+            // eslint-disable-next-line no-console
+            console.info('Animation ', message, ' : ', animation.name);
+        }
+    };
+} else {
+    _DEBUG = function _DEBUG() {};
+}
+
+// Private functions
+// stop timer and re-init parameter
+const resetTimer = function resetTimer() {
+    if (this.id) {
+        clearInterval(this.id);
+        this.id = undefined;
+    }
+
+    this.keyframe = 0;
+};
+
+// finish animation and re-init parameter
+const finishAnimation = function finishAnimation() {
+    resetTimer.bind(this)();
+    this.animation = null;
+    this.callBackFrame = () => {};
+    if (this.callBack) {
+        this.callBack();
+        this.callBack = null;
+    }
+    this.resolve();
+};
+
+const setPlayerState = function setPlayerState(state) {
+    this.state = state;
+    _DEBUG(debugMsg[state], this.animation);
+};
+
+const frameEvent = new CustomEvent('frameAnimation');
+const stopEvent = new CustomEvent('stopAnimation');
+const endEvent = new CustomEvent('endAnimation');
+
+/**
+ * AnimationPlayer
+ * It can play, pause or stop animation
+ */
+class AnimationPlayer {
+    constructor(dom) {
+        this.dom = dom;
+        this.id = null;
+        this.keyframe = 0;
+        this.animation = null;
+        this.resolve = null;
+        this.promise = new Promise(r => this.resolve = r);
+        this.state = PLAYER_STATE.STOP;
+        this.callBack = null;
+        this.callBackFrame = () => {};
+    }
+
+    /**
+     * Gets the interval time.
+     *
+     * @return {Number}  The interval time is the time between consecutive frames
+     */
+    static get intervalTime() {
+        return INTERVAL_TIME;
+    }
+
+    isPlayed() {
+        return this.state === PLAYER_STATE.PLAY;
+    }
+
+    isStopped() {
+        return this.state === PLAYER_STATE.STOP;
+    }
+
+    isEnded() {
+        return this.state === PLAYER_STATE.END;
+    }
+
+    // Public functions
+
+    /**
+     * { Start animation }
+     *
+     * @param      {Animation} The animation to play
+     * @return     {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    play(animation, callBack) {
+        this.callBack = callBack;
```

Comment:
> I suggested to use callback for each frame

This has been the case since the beginning, [it was for callback in animation's ending](https://github.com/iTowns/itowns2/pull/218#discussion_r92003633). I see this